src/content/containers/Home.js

```diff
@@ -33,6 +33,27 @@ type InstallButtonProps = {
   className?: string,
 };
 
+type UploadButtonProps = {
+  retrieveProfileFromFile: File => void,
+};
+
+class UploadButton extends PureComponent {
+  props: UploadButtonProps;
+
+  render() {
+    return (
+      <div>
+        <input type="file" ref="file"></input>
+        <button onClick={e => this._upload()}>Upload</button>
```

Comment:
nit: This is not a big deal in this case, but it's better to avoid inline functions for props. Instead you can bind `_upload` in the constructor like we do for example in the Home component below.